rotkehlchen/chain/ethereum/modules/cowswap/decoder.py

```diff
@@ -0,0 +1,236 @@
+                    token_type=EvmTokenKind.ERC20,
+                ),
+            )
+            from_amount = asset_normalized_value(amount=raw_from_amount, asset=from_asset)
+            to_amount = asset_normalized_value(amount=raw_to_amount, asset=to_asset)
+
+            trades.append(TradeData(
+                from_asset=from_asset,
+                from_amount=from_amount,
+                to_asset=to_asset,
+                to_amount=to_amount,
+            ))
+
+        return trades
+
+    def _detect_relevant_trades(
+            self,
+            trades: list[TradeData],
+            decoded_events: list[HistoryBaseEntry],
+    ) -> tuple[list[tuple[TradeData, ChecksumEvmAddress]], list[HistoryBaseEntry]]:
+        """
+        This function does the following
+        1. Detect trades that are relevant to tracked accounts.
+        2. For each trade, find the corresponding tracked account.
+
+        Each relevant trade has an optional spend event (since sometimes ETH is spent,
+        and in these cases there is no spend event) and a mandatory receive event.
+
+        So to detect relevant trades we check which trades have a decoded eth/token receive event.
+        """
+        receive_events: dict[tuple[Asset, FVal], ChecksumEvmAddress] = {}
+        other_events = []  # other events such as approvals
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.counterparty == GPV2_SETTLEMENT_ADDRESS
+            ):
+                receive_events[(event.asset, event.balance.amount)] = event.location_label  # type: ignore[assignment]  # location label is not None here  # noqa: E501
+            elif (
+                event.event_type != HistoryEventType.SPEND or
+                event.counterparty != GPV2_SETTLEMENT_ADDRESS
+            ):
+                other_events.append(event)
+
+        relevant_trades: list[tuple[TradeData, ChecksumEvmAddress]] = []  # trades and traders addresses  # noqa: E501
+        for trade in trades:
+            trader_address = receive_events.get((trade.to_asset, trade.to_amount))
+            if trader_address is not None:
+                relevant_trades.append((trade, trader_address))
+
+        return relevant_trades, other_events
+
+    def _aggregator_post_decoding(
+            self,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],
+    ) -> list[HistoryBaseEntry]:
+        if transaction.to_address != GPV2_SETTLEMENT_ADDRESS:
+            return decoded_events
+
+        all_trades = self._find_trades(all_logs)
+        relevant_trades, other_events = self._detect_relevant_trades(all_trades, decoded_events)
+
+        new_events = []
+        for trade, trader_address in relevant_trades:
+            # Generate a spend event
+            new_events.append(self.base.make_event_next_index(
```

Comment:
I don't understand here. Why are you making new events? Aren't the token transfers already seen by rotki? You are adding more events on top of them here.

Reply:
Changed it